### PR TITLE
Update docs for agent version 3.34.1

### PR DESCRIPTION
**Pull request by moskyb**
This adds the `--reject-secrets` flag and the `BUILDKITE_AGENT_PIPELINE_UPLOAD_REJECT_SECRETS` envar for the `buildkite-agent pipline upload` command

### CODE REVIEW

File: pages/agent/v3/help/_pipeline_upload.md [hunk 1]
```diff
@@ -50,6 +50,7 @@ scripts that generate dynamic pipelines.
 <tr id="job"><th><code>--job value</code> <a class="Docs__attribute__link" href="#job">#</a></th><td><p>The job that is making the changes to its build<br /><strong>Environment variable</strong>: <code>$BUILDKITE_JOB_ID</code></p></td></tr>
 <tr id="dry-run"><th><code>--dry-run </code> <a class="Docs__attribute__link" href="#dry-run">#</a></th><td><p>Rather than uploading the pipeline, it will be echoed to stdout<br /><strong>Environment variable</strong>: <code>$BUILDKITE_PIPELINE_UPLOAD_DRY_RUN</code></p></td></tr>
 <tr id="no-interpolation"><th><code>--no-interpolation </code> <a class="Docs__attribute__link" href="#no-interpolation">#</a></th><td><p>Skip variable interpolation the pipeline when uploaded<br /><strong>Environment variable</strong>: <code>$BUILDKITE_PIPELINE_NO_INTERPOLATION</code></p></td></tr>
+<tr id="reject-secrets"><th><code>--reject-secrets </code> <a class="Docs__attribute__link" href="#reject-secrets">#</a></th><td><p>When true, fail the pipeline upload early if the the pipeline contains secrets<br /><strong>Environment variable</strong>: <code>$BUILDKITE_AGENT_PIPELINE_UPLOAD_REJECT_SECRETS</code></p></td></tr>
```
Comment (waferbaby) [inline pages/agent/v3/help/_pipeline_upload.md:53]: Typo: "the the"

Comment (moskyb) [reply pages/agent/v3/help/_pipeline_upload.md:53]: oh shit, that comes from the [agent code](https://github.com/buildkite/agent/blob/main/clicommand/pipeline_upload.go#L103) 🙈

sigh. i'll include a fix in 3.35 when it goes through tomorrow.

Comment (pda) [reply pages/agent/v3/help/_pipeline_upload.md]: I've added a quick fix to this PR;

> (This change will be overwritten next time we regenerate the file, but @moskyb will fix the source before that 😁)

Also I made a typo in the commit message 🎉

**Comment by pda** [comment]
image

Looks good in the test deploy 👍🏼

**Review by pda — APPROVED** [review]

**Comment by plaindocs** [comment]
Cheers folks!

**Pull request merged**